Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/executing/QueryEvaluator.java

```diff
@@ -74,24 +81,35 @@ private static boolean evaluateWhere(Query elm, State state, ElmLibraryVisitor<O
         return true;
     }
 
-    private static Object evaluateReturn(Query elm, State state, List<Variable> variables, List<Object> elements, ElmLibraryVisitor<Object, State> visitor) {
-        return elm.getReturn() != null ? visitor.visitExpression(elm.getReturn().getExpression(), state) : constructResult(state, variables, elements);
+    private static Object evaluateReturn(Query elm, State state, List<Variable> variables, List<Object> elements,
+            ElmLibraryVisitor<Object, State> visitor) {
+        return elm.getReturn() != null ? visitor.visitExpression(elm.getReturn().getExpression(), state)
+                : constructResult(state, variables, elements);
+    }
+
+    private static List<Object> evaluateAggregate(AggregateClause elm, State state, ElmLibraryVisitor<Object, State> visitor, List<Object> elements) {
+        return Collections.singletonList(AggregateClauseEvaluator.aggregate(elm, state, visitor, elements));
+    }
+
+    private static Object constructTuple(State state, List<Variable> variables, List<Object> elements) {
+        LinkedHashMap<String, Object> elementMap = new LinkedHashMap<>();
+        for (int i = 0; i < variables.size(); i++) {
```

Comment:
Just curious why you didn't each a for-each here.

Reply:
Copy-paste. I'll fix.